### PR TITLE
[tune] Safer try-catch for TensorboardX

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?
TensorboardX doesn't support many types for its logger.

This prevents the user 'error'.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/logger.py [hunk 1]
```diff
@@ -186,6 +186,8 @@ class TBXLogger(Logger):
         {"a": {"b": 1, "c": 2}} -> {"a/b": 1, "a/c": 2}
     """
 
+    VALID_HPARAMS = {str, bool, int, float, None}
```
Comment (hartikainen) [inline python/ray/tune/logger.py:189]: Should this include `list`?
```suggestion
    VALID_HPARAMS = {str, bool, int, float, list, None}
```

File: python/ray/tune/logger.py [hunk 2]
```diff
@@ -253,14 +255,29 @@ def _try_log_hparams(self, result):
         flat_params = flatten_dict(self.trial.evaluated_params)
         scrubbed_params = {
             k: v
-            for k, v in flat_params.items() if v is not None
+            for k, v in flat_params.items() if type(v) in VALID_SUMMARY_TYPES
```
Comment (hartikainen) [inline python/ray/tune/logger.py:258]:
```suggestion
            for k, v in flat_params.items() if isinstance(v, VALID_SUMMARY_TYPES)
```

File: python/ray/tune/logger.py [hunk 3]
```diff
@@ -253,14 +255,29 @@ def _try_log_hparams(self, result):
...
+        }
+
+        removed = {
+            k: v
+            for k, v in flat_params.items()
+            if type(v) not in VALID_SUMMARY_TYPES
```
Comment (hartikainen) [inline python/ray/tune/logger.py:264]:
```suggestion
            if not isinstance(v, VALID_SUMMARY_TYPES)
```

File: python/ray/tune/logger.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/logger.py:189]: Does list actually work?

**Review by hartikainen — APPROVED** [review]

**Pull request merged**